cdap-common/bin/common.sh

```diff
@@ -270,6 +270,22 @@ cdap_set_hive_classpath() {
   fi
 }
 
+# Get SPARK_HOME
+cdap_set_spark() {
+  # First, see if we're set to something sane
+  if [ -n "${SPARK_HOME}" -a -d "${SPARK_HOME}" ]; then
+    return 0 # SPARK_HOME is set, already
+  else
+    if [[ $(which spark-shell 2>/dev/null) ]]; then
+      SPARK_VAR_OUT=$(echo 'for ((key, value) <- sys.env) println (key + "=" + value); exit' | spark-shell --master local 2>/dev/null)
+      SPARK_HOME=$(echo -e "${SPARK_VAR_OUT}" | grep ^SPARK_HOME= | cut -d= -f2)
+      export SPARK_HOME
```

Comment:
I can do that. I didn't find it necessary since we're parsing the data from running Spark, itself. The call wouldn't work if it was misconfigured.